src/async-wrap.cc

```diff
@@ -116,13 +116,14 @@ static void DisableHooksJS(const FunctionCallbackInfo<Value>& args) {
 static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
-  CHECK(args[0]->IsFunction());
-  CHECK(args[1]->IsFunction());
-  CHECK(args[2]->IsFunction());
-
-  env->set_async_hooks_init_function(args[0].As<Function>());
-  env->set_async_hooks_pre_function(args[1].As<Function>());
-  env->set_async_hooks_post_function(args[2].As<Function>());
+  if (args[0]->IsFunction())
+    env->set_async_hooks_init_function(args[0].As<Function>());
```

Comment:
Since the init callback is always necessary, will change this to fail if init callback is not passed.